core/domain/user_query_jobs_one_off_test.py

```diff
@@ -0,0 +1,157 @@
+    EXP_ID_1 = 'exp_id_1'
+    EXP_ID_2 = 'exp_id_2'
+    USER_A_EMAIL = 'a@example.com'
+    USER_A_USERNAME = 'a'
+    USER_B_EMAIL = 'b@example.com'
+    USER_B_USERNAME = 'b'
+    USER_C_EMAIL = 'c@example.com'
+    USER_C_USERNAME = 'c'
+    USER_D_EMAIL = 'd@example.com'
+    USER_D_USERNAME = 'd'
+    USER_SUBMITTER_EMAIL = 'submit@example.com'
+    USER_SUBMITTER_USERNAME = 'submit'
+
+    def _run_one_off_job(self, query_id):
+        """Runs the one-off MapReduce job."""
+        job_id = user_query_jobs_one_off.UserQueryOneOffJob.create_new()
+        params = {
+            'query_id': query_id
+        }
+        user_query_jobs_one_off.UserQueryOneOffJob.enqueue(
+            job_id, additional_job_params=params)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT), 1)
+        self.process_and_flush_pending_tasks()
+
+    def setUp(self):
+        super(UserQueryJobOneOffTests, self).setUp()
+        # User A has no created or edited explorations
+        # User B has one created exploration
+        # User C has one edited exploration
+        # User D has created an exploration and then edited it.
+        # (This is used to check that there are no duplicate
+        # entries in the contribution lists.)
+        # Submitter is the user who submits the query.
+        self.signup(self.USER_A_EMAIL, self.USER_A_USERNAME)
+        self.user_a_id = self.get_user_id_from_email(self.USER_A_EMAIL)
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        self.user_b_id = self.get_user_id_from_email(self.USER_B_EMAIL)
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        self.user_c_id = self.get_user_id_from_email(self.USER_C_EMAIL)
+        self.signup(self.USER_D_EMAIL, self.USER_D_USERNAME)
+        self.user_d_id = self.get_user_id_from_email(self.USER_D_EMAIL)
+        self.signup(self.USER_SUBMITTER_EMAIL, self.USER_SUBMITTER_USERNAME)
+        self.submitter_id = self.get_user_id_from_email(
+            self.USER_SUBMITTER_EMAIL)
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_1, self.user_b_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_c_id, self.EXP_ID_1, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_2, self.user_d_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_d_id, self.EXP_ID_2, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+        # set tmpsuperadm1n as admin in ADMIN_USERNAMES config property.
+        self.set_admins(['tmpsuperadm1n'])
+
+    def test_user_is_active_in_last_n_days(self):
```

Comment:
To really test this, the set of users should also include someone who created an exp more than N days ago.